### PR TITLE
[Core] minor cleanup SubModelPartSkinDetectionProcess

### PR DESCRIPTION
**Pull request by philbucher**
**Description**
Minor refactor
- moving impl to cpp
- using varibale utils

### CODE REVIEW

File: kratos/processes/sub_model_part_skin_detection_process.cpp [hunk 1]
```diff
@@ -12,10 +12,28 @@
...
 #include "sub_model_part_skin_detection_process.h"
+#include "utilities/variable_utils.h"
 
 namespace Kratos
 {
 
+template<SizeType TDim>
+void SubModelPartSkinDetectionProcess<TDim>::SelectIfAllNodesOnSubModelPart::Prepare(ModelPart& rMainModelPart) const
+{
+    VariableUtils().SetFlag(SubModelPartSkinDetectionProcess::NODE_SELECTED, true, rMainModelPart.GetSubModelPart(mName).Nodes());
```
Comment (philbucher) [inline kratos/processes/sub_model_part_skin_detection_process.cpp:23]: @roigcarlo 
`Set(SOME_FLAG)` is the same as `Set(SOME_FLAG, true)` right?

File: kratos/processes/sub_model_part_skin_detection_process.cpp [hunk 2]
```diff
@@ -12,10 +12,28 @@
...
+}
+
+template<SizeType TDim>
+bool SubModelPartSkinDetectionProcess<TDim>::SelectIfAllNodesOnSubModelPart::IsSelected(const Geometry<Node<3>>::PointsArrayType& rNodes) const
+{
+    bool select = true;
+    for (auto i_node = rNodes.begin(); i_node != rNodes.end(); ++i_node)
```
Comment (loumalouomega) [inline kratos/processes/sub_model_part_skin_detection_process.cpp]:
```suggestion
    for (auto i_node = rNodes.begin(); i_node != rNodes.end(); ++i_node) {
```

File: kratos/processes/sub_model_part_skin_detection_process.cpp [hunk 3]
```diff
@@ -12,10 +12,28 @@
...
+    {
```
Comment (loumalouomega) [inline kratos/processes/sub_model_part_skin_detection_process.cpp]:
```suggestion
```

**Review by loumalouomega — COMMENTED** [review]
Minor comments

File: kratos/processes/sub_model_part_skin_detection_process.cpp (hunk 1 above)
Comment (philbucher) [reply kratos/processes/sub_model_part_skin_detection_process.cpp:23]: checked myself, confirming

**Review by rubenzorrilla — APPROVED** [review]

**Pull request merged**